src/server/game/Entities/Object/ObjectGuid.h

```diff
@@ -233,18 +277,33 @@ class PackedGuid
         ByteBuffer _packedGuid;
 };
 
-template<HighGuid high>
-class ObjectGuidGenerator
+
+class ObjectGuidGeneratorBase
 {
-    public:
-        explicit ObjectGuidGenerator(uint32 start = 1) : _nextGuid(start) { }
+public:
+    ObjectGuidGeneratorBase(ObjectGuid::LowType start = 1) : _nextGuid(start) { }
 
-        void Set(uint32 val) { _nextGuid = val; }
-        uint32 Generate();
-        uint32 GetNextAfterMaxUsed() const { return _nextGuid; }
+    virtual void Set(uint32 val) { _nextGuid = val; }
+    virtual ObjectGuid::LowType Generate() = 0;
+    ObjectGuid::LowType GetNextAfterMaxUsed() const { return _nextGuid; }
 
-    private:
-        uint32 _nextGuid;
+protected:
+    static void HandleCounterOverflow(HighGuid high);
+    uint64 _nextGuid;
```

Comment:
From what I can see it happened here

https://github.com/TrinityCore/TrinityCore/commit/9e1930959db59013d6f0221d29cc652cdf2f6145

I suspect you had to change them all from uint32 to uint64 anyway and decided it made more sense to just change to use the LowType instead. Which I agree with.

But it was done before (and separately) hence I never picked it up.

Is it the intention this should be in 3.3.5? I mean I agree it should be, but it's not an urgency. I made a branch for it anyway. But I think it's low priority. Agree?